core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -264,24 +291,87 @@ export class TranslationSuggestionReviewModalComponent implements OnInit {
     const response = await this.threadDataBackendApiService.fetchMessagesAsync(
       threadId);
     const threadMessageBackendDicts = response.messages;
-    this.reviewMessage = threadMessageBackendDicts.map(
-      m => ThreadMessage.createFromBackendDict(m))[1].text;
+    let threadMessages = threadMessageBackendDicts.map(
+      m => ThreadMessage.createFromBackendDict(m));
+    if (threadMessages[1] !== undefined) {
+      this.reviewMessage = threadMessages[1].text;
+    }
   }
 
-  showNextItemToReview(suggestionId: string): void {
-    this.resolvedSuggestionIds.push(this.activeSuggestionId);
-    if (this.lastSuggestionToReview) {
+  gotoNextItem(): void {
+    // If the current item is the last item, do not navigate.
+    if (this.isLastItem) {
+      return;
+    }
+    // This prevents resolved contributions from getting added to the list.
+    if (!this.resolvedSuggestionIds.includes(this.activeSuggestionId)) {
+      this.skippedContributionIds.push(this.activeSuggestionId);
+    }
+
+    let lastContributionId = (
+      this.remainingContributionIds[this.remainingContributionIds.length - 1]);
+
+    this.activeSuggestionId = lastContributionId;
+    this.activeContribution = this.allContributions[
+      lastContributionId];
+
+    this.remainingContributionIds.pop();
+
+    if (this.remainingContributionIds.length === 0) {
+      this.isLastItem = true;
+    }
+
+    if (this.skippedContributionIds.length === 0) {
+      this.isFirstItem = true;
+    } else {
+      this.isFirstItem = false;
+    }
+
+    // Close modal instance if the suggestion's corresponding opportunity
+    // is deleted. See issue #14234.
+    if (!this.activeContribution.details) {
       this.activeModal.close(this.resolvedSuggestionIds);
       return;
     }
 
-    let lastContribution = (
-      Object.keys(this.remainingContributions)[
-        Object.keys(this.remainingContributions).length - 1]);
-    this.activeSuggestionId = lastContribution;
-    this.activeContribution = this.remainingContributions[
-      lastContribution];
-    delete this.remainingContributions[this.activeSuggestionId];
+    this.activeSuggestion = this.activeContribution.suggestion;
+    this.activeContributionDetails = this.activeContribution.details;
+    this.contextService.setCustomEntityContext(
+      AppConstants.IMAGE_CONTEXT.EXPLORATION_SUGGESTIONS,
+      this.activeSuggestion.target_id);
+    this.subheading = `${this.activeContributionDetails.topic_name} / ` +
+      `${this.activeContributionDetails.story_title} / ` +
+      `${this.activeContributionDetails.chapter_title}`;
+    this.init();
```

Comment:
Why is this called?

Reply:
The init() function is called to refresh the modal content with the new suggestion data (after navigating)

Reply:
That is a bit confusing, can you rename the function to smth like refresh and then call it from the init and from here too?

Reply:
Not done?

Reply:
No, do not call init from here, that is in my opinion the confusing thing. Rather move everything from init to the refresh and then call refresh from everywhere.

Reply:
Not done.